Fix integration tests

Anviking opened the pull request:
# Issue
None, but bors was failing for unrelated reasons in #764, and I was able to reproduce the same failures locally.

# Overview

- [x] Allow ports up to `65535` in CLI (we might generate them randomly https://input-output-rnd.slack.com/archives/GBT05825V/p1569852386030200)


# Comments

<!-- Additional comments or screenshots to attach if any -->

<!-- 
Don't forget to:

 ✓ Self-review your changes to make sure nothing unexpected slipped through
 ✓ Assign yourself to the PR
 ✓ Assign one or several reviewer(s)
 ✓ Once created, link this PR to its corresponding ticket
 ✓ Acknowledge any changes required to the Wiki
-->

Anviking committed:
Add HasCallStack to shouldContainT

Anviking committed:
Update incorrect comment for collectStreams

Anviking committed:
Allow larger range of ports from CLI

Anviking commented on lib/core-integration/src/Test/Integration/Framework/DSL.hs:
```diff
@@ -1243,7 +1245,7 @@ collectStreams (nOut0, nErr0) p = do
         hSetBuffering e LineBuffering
         let io = race
-                (threadDelay (65 * oneSecond))
+                (threadDelay (30 * oneSecond))
```
Not totally sure on this though. Maybe the default `waitForConnection` should have a chance to timeout.

KtorZ replied:
Not sure this has anything to do with the CI failures :thinking:

Anviking replied:
I was lead to believe that, but doesn't seem like that now, no.

Anviking commented on lib/jormungandr/test/integration/Test/Integration/Jormungandr/Scenario/CLI/Launcher.hs:
```diff
@@ -150,8 +150,9 @@ spec = do
+            (Exit c, Stdout o, Stderr e) <- cardanoWalletCLI @t args
             c `shouldBe` ExitFailure 1
+            e `shouldBe` ""
             o `shouldContain` ("As far as I can tell, this isn't a valid block file: " <> secret)
```
Suspicious: In this test we check for the message in stdout. The other tests check in stderr.

KtorZ replied:
CLI argument parsing failures are sent on stderr; then once the application has started, it sends its logs on stdout. In this case, this happens when the application has started though it should probably happens before.

rvl commented on lib/cli/src/Cardano/CLI.hs:
```diff
@@ -1022,12 +1022,9 @@ newtype Port (tag :: Symbol) = Port { getPort :: Int }
 instance Bounded (Port tag) where
     minBound = Port 1024
-    maxBound = Port 49151
+    maxBound = Port 65535
```
Seems totally reasonable to use the full range of ports. If the OS selects a port then it's a good port to use.

The minBound could also be 0, because a. the wallet could be connecting to a node listening on any port, and b. in some contexts such as a docker container, the wallet may well be running as "root" and thus be able to listen on any available port. But I'm not sure how bad the exception message is when the port can't be used, so maybe leave it.

Anviking committed:
Remove hSetEncoding in http-bridge integration tests

It could be why integration tests are failing with

  src/Test/Integration/Scenario/CLI/Mnemonics.hs:30:5:
      1) Mnemonics CLI tests CLI_MNEMONICS_01 - Can generate mnemonics with default size
           uncaught exception: IOException of type InvalidArgument
           fd:18: hGetContents: invalid argument (invalid byte sequence)

on

stack test cardano-wallet-http-bridge:integration --ta '--match "/Mnemonics CLI tests/CLI_MNEMONICS_01 - Can generate mnemonics with default size/"'

KtorZ committed:
make sure to flush out logs before exiting

KtorZ committed:
remove mnemonic integration tests from http-bridge

We've already spent way too much time trying to figure out this. No one can reproduce locally. This is
a wasted effort for something we're going to throw away. Plus, this is already tested in jormungandr
